curp/src/members.rs

```diff
@@ -0,0 +1,200 @@
+use std::{
+    collections::{hash_map::DefaultHasher, HashMap},
+    hash::Hasher,
+    time::SystemTime,
+};
+
+use itertools::Itertools;
+
+use crate::ServerId;
+
+/// cluster members information
+#[derive(Debug)]
+pub struct ClusterMember {
+    /// current server id
+    id: String,
+    /// other peers information
+    all_members: HashMap<ServerId, String>,
+}
+
+impl ClusterMember {
+    /// Construct a new `ClusterMember`
+    ///
+    /// # Panics
+    ///
+    /// panic if `all_members` is empty
+    #[inline]
+    #[must_use]
+    pub fn new(all_members: HashMap<ServerId, String>, id: String) -> Self {
+        assert!(!all_members.is_empty());
+        Self { id, all_members }
+    }
+
+    /// get server addres via server id
+    #[must_use]
+    #[inline]
+    pub fn get_address(&self, id: &str) -> Option<&String> {
```

Comment:
Should we return an `Option<&str>` instead of an `Option<&String>` for more flexibility?

Reply:
What's the meaning of the flexibility you mentioned here? Any example?

Reply:
Done